### PR TITLE
Removed coercion of `inout` to temporary pointer

### PR DESCRIPTION
**Pull request by regexident**
Fixes https://github.com/Jounce/Surge/issues/162.

### CODE REVIEW

File: Sources/Surge/Digital Signal Processing/FFT.swift [hunk 1]
```diff
@@ -25,57 +25,73 @@ import Accelerate
...
-    var splitComplex = DSPDoubleSplitComplex(realp: &real, imagp: &imaginary)
 
-    let length = vDSP_Length(floor(log2(Float(input.count))))
-    let radix = FFTRadix(kFFTRadix2)
-    let weights = vDSP_create_fftsetupD(length, radix)
-    withUnsafeMutablePointer(to: &splitComplex) { splitComplex in
-        vDSP_fft_zipD(weights!, splitComplex, 1, length, FFTDirection(FFT_FORWARD))
-    }
+    return real.withUnsafeMutableBufferPointer { realBuffer in
+    imaginary.withUnsafeMutableBufferPointer { imaginaryBuffer in
```
Comment (mattt) [inline Sources/Surge/Digital Signal Processing/FFT.swift:67]:
```suggestion
    return real.withUnsafeMutableBufferPointer { realBuffer in
        imaginary.withUnsafeMutableBufferPointer { imaginaryBuffer in
```

**Comment by mattt** [comment]
Looks like CI is being flakey for some reason, reporting failure when another run for the same commit passed. Might be a good reason to migrate over to GitHub Actions. I can take that on, if you're interested, @regexident.

Otherwise, this all looks good to me.

**Comment by regexident** [comment]
> I can take that on, if you're interested, @regexident.

If you don't mind that would be great!

**Comment by mattt** [comment]
Merging master into PR branch to re-run tests against new CI infrastructure.

**Pull request merged**